src/anomalib/data/base/datamodule.py

```diff
@@ -227,3 +227,41 @@ def test_dataloader(self) -> EVAL_DATALOADERS:
     def predict_dataloader(self) -> EVAL_DATALOADERS:
         """Use the test dataloader for inference unless overridden."""
         return self.test_dataloader()
+
+    @property
+    def user_transform(self) -> Transform:
+        """This property returns the user-specified transform, if any.
+
+        This property is accessed by the engine to set the transform for the model. The eval_transform takes precedence
+        over the train_transform, because the transform that we store in the model is the one that should be used during
+        inference.
+        """
+        if self._eval_transform:
+            return self._eval_transform
+        if self._train_transform:
+            return self.train_transform
+        return None
+
+    @property
+    def train_transform(self) -> Transform:
+        """Get the transforms that the datamodule should apply during training.
+
+        If the train_transform is not set, the engine will request the transform from the model.
+        """
+        if self._train_transform:
+            return self._train_transform
+        if self.trainer and self.trainer.model:
```

Comment:
Does this work if `trainer` is not available? If not, would it be safer to check if via hasattr(self, "trainer")

Reply:
good point, missed that one